### PR TITLE
cleanup(clowder): unify puma config and simplify entrypoint

### PR DESCRIPTION
**Pull request by skateman**
Felt cute, might delete later :grin: 

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -1,45 +1,53 @@
...
+# Specifies the `host` that puma will listen on to receive requests
 #
-threads_count = ENV.fetch('PUMA_MAX_THREADS') { 5 }
-threads threads_count, threads_count
+set_default_host '0.0.0.0'
 
 # Specifies the `port` that Puma will listen on to receive requests; default is 3000.
 #
-port Settings.web_port
+port ClowderCommonRuby::Config.load.webPort
```
Comment (vkrizan) [inline config/puma.rb:14]: Are there any risk when calling the the clowder config load here?
Weren't the clowder config load integrated with `Settings`?

**Review by vkrizan — APPROVED** [review]
:scissors: :+1:

**Comment by vkrizan** [comment]
@skateman this does not seem healthy :grin::
```
/opt/app-root/src/.bundle/ruby/3.1.0/gems/puma-6.0.2/lib/puma/launcher.rb:304:in `write': No such file or directory @ rb_sysopen - tmp/pids/server.pid (Errno::ENOENT)
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/puma-6.0.2/lib/puma/launcher.rb:304:in `write_pid'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/puma-6.0.2/lib/puma/launcher.rb:112:in `write_state'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/puma-6.0.2/lib/puma/cluster.rb:408:in `run'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/puma-6.0.2/lib/puma/launcher.rb:186:in `run'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/puma-6.0.2/lib/puma/cli.rb:75:in `run'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/puma-6.0.2/bin/puma:10:in `<top (required)>'
	from /opt/app-root/src/.bundle/ruby/3.1.0/bin/puma:25:in `load'
	from /opt/app-root/src/.bundle/ruby/3.1.0/bin/puma:25:in `<top (required)>'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/cli/exec.rb:58:in `load'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/cli/exec.rb:23:in `run'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/cli.rb:486:in `exec'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/cli.rb:31:in `dispatch'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/cli.rb:25:in `start'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/exe/bundle:48:in `block in <top (required)>'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'
	from /opt/app-root/src/.bundle/ruby/3.1.0/gems/bundler-2.3.26/exe/bundle:36:in `<top (required)>'
	from /usr/bin/bundle:25:in `load'
	from /usr/bin/bundle:25:in `<main>'
```

File: config/puma.rb [hunk 2]
```diff
@@ -1,45 +1,53 @@
...
 
 # Specifies the `environment` that Puma will run in.
 #
-environment ENV.fetch('RAILS_ENV') { 'development' }
+environment ENV.fetch('RAILS_ENV', 'development')
 
 # Specifies the `pidfile` that Puma will use.
-pidfile ENV.fetch("PIDFILE") { "tmp/pids/server.pid" }
+pidfile ENV.fetch('PIDFILE', 'tmp/pids/server.pid')
```
Comment (vkrizan) [inline config/puma.rb:21]: https://github.com/RedHatInsights/compliance-backend/pull/1709#issuecomment-1611270571

Comment (skateman) [reply config/puma.rb:21]: This is not the problem, the `pids` folder doesn't exist and it cannot be created :disappointed:

File: config/puma.rb (hunk 1 above)
Comment (skateman) [reply config/puma.rb:14]: When puma loads, clowder config isn't available

**Comment by skateman** [comment]
Somehow puma's config is interfering with the yabeda metrics server in sidekiq/racecar...will take a look later.

**Pull request merged**